### PR TITLE
Add new parameters for sealights hook

### PR DESCRIPTION
**Pull request by CubikNeRubik**
Thanks for contributing to the buildpack. To speed up the process of reviewing your pull request please provide us with:

* A short explanation of the proposed change:
I have update sealight build hook to support installing of specific version

* An explanation of the use cases your change solves
User can set extra parameters in sealights user-provided services as `version` and `customAgentUrl`
Also recomended version from sealight server can be requested if `SL_DOMAIN` was provided

* [x] I have viewed signed and have submitted the Contributor License Agreement

* [x] I have made this pull request to the `develop` branch

* [ ] I have added an integration test

### CODE REVIEW

File: src/nodejs/hooks/sealights.go [hunk 1]
```diff
@@ -346,62 +402,140 @@ func (sl *SealightsHook) injectSealights(stager *libbuildpack.Stager) error {
...
+			queryString := func(key string) string {
+				if value, ok := service.Credentials[key].(string); ok {
+					return value
+				}
+				return ""
+			}
+
+			options := &SealightsParameters{
+				Token:          queryString("token"),
+				TokenFile:      queryString("tokenFile"),
```
Comment (yishayh) [inline src/nodejs/hooks/sealights.go:439]: @CubikNeRubik we have to have the `labId`, `buildSessionIdFile` and `buildSessionId` as part of this release.

File: src/nodejs/hooks/sealights.go [hunk 2]
```diff
@@ -149,13 +198,19 @@ func (sl *SealightsHook) usePackageJson(originalStartCommand string, stager *lib
 	return fmt.Errorf(NpmCommandStringError), false
 }
 
-func (sl *SealightsHook) getSealightsOptions(app string, token string, tokenFile string) *SealightsOptions {
-	o := &SealightsOptions{
-		Token:       token,
-		TokenFile:   tokenFile,
+func (sl *SealightsHook) getSealightsOptions(app string) *SealightsRunOptions {
+
+	proxy := os.Getenv("SL_PROXY")
```
Comment (yishayh) [inline src/nodejs/hooks/sealights.go:203]: @CubikNeRubik like you said, please add the following here:

		BsId:        os.Getenv("SL_BUILD_SESSION_ID"),
		BsIdFile:    os.Getenv("SL_BUILD_SESSION_ID_FILE")
		LabId:       os.Getenv("SL_LAB_ID"),
		ProjectRoot: os.Getenv("SL_PROJECT_ROOT"),
		TestStage:   os.Getenv("SL_TEST_STAGE"),

In the same format like `SL_PROXY` and `sl.parameters.Proxy`

File: src/nodejs/hooks/sealights.go [hunk 3]
```diff
@@ -58,17 +95,28 @@ type PackageJson struct {
...
+		Command:     command,
+		HttpClient:  httpClient,
+		parameters:  &SealightsParameters{},
+		initialized: false,
+	}
 }
 
 func (sl *SealightsHook) AfterCompile(stager *libbuildpack.Stager) error {
-	sl.Log.Info("inside Sealights hook")
+	sl.Log.Debug("inside Sealights hook")
```
Comment (yishayh) [inline src/nodejs/hooks/sealights.go:114]: Please remove the debug logs.

File: src/nodejs/hooks/sealights.go [hunk 4]
```diff
@@ -346,62 +402,140 @@ func (sl *SealightsHook) injectSealights(stager *libbuildpack.Stager) error {
...
 }
 
-func containsSealightsService(key string, services interface{}, query string) bool {
-	var serviceName string
+func (sl *SealightsHook) parseVcapServices() {
 
-	if strings.Contains(key, query) {
-		return true
+	if sl.initialized {
+		sl.Log.Debug("already initialized. config won`t be parsed")
```
Comment (yishayh) [inline src/nodejs/hooks/sealights.go:408]: Please remove the debug logs

Comment (CubikNeRubik) [reply src/nodejs/hooks/sealights.go:408]: This log not shown by default
user should configure Debug mode to see that

**Review by yishayh — APPROVED** [review]

**Review by brayanhenao — APPROVED** [review]

**Pull request merged**